src/liballoc/collections/btree/node.rs

```diff
@@ -240,7 +240,7 @@ impl<K, V> Root<K, V> {
     /// Adds a new internal node with a single edge, pointing to the previous root, and make that
     /// new node the root. This increases the height by 1 and is the opposite of `pop_level`.
     pub fn push_level(&mut self) -> NodeRef<marker::Mut<'_>, K, V, marker::Internal> {
-        debug_assert!(!self.is_shared_root());
+        debug_assert!(!self.is_shared_root()); // &mut to a static is impossible anyway
```

Comment:
I would like to see us drop the debug asserts that should be statically impossible (e.g. like here, where we have unique access).

Reply:
Dropped

Reply:
So actually, not committing that (yet). Seems I should retract the comment instead. And perhaps mark unsafe. Or do you imply that `&mut self` _logically_ implies that the caller should know that the shared root is unacceptable?

Reply:
Yes, we should retract this. I missed that we don't actually "own" the root here necessarily (i.e., we may have a NonNull<Shared> or perhaps not)